framework/src/mesh/GeneratedMesh.C

```diff
@@ -43,6 +43,8 @@ InputParameters validParams<GeneratedMesh>()
   params.addParam<Real>("zmax", 1.0, "Upper Z Coordinate of the generated mesh");
   params.addParam<MooseEnum>("elem_type", elem_types, "The type of element from libMesh to generate (default: linear element for requested dimension)");
 
+  params.addParam<std::vector<SubdomainID> >("subdomain_ids", "Subdomain IDs of all generated elements (refer to libMesh::MeshTools::Generation for element ordering");
+
```

Comment:
Wait, you really want to supply all the IDs for every element? While this is the most flexible it does seem all that useful. Even a 10x10 mesh would require 100 entries in your vector.